common/djangoapps/student/models.py

```diff
@@ -1002,7 +1005,6 @@ def __unicode__(self):
         ).format(self.user, self.course_id, self.created, self.is_active)
 
     @classmethod
-    @transaction.atomic
```

Comment:
Although I am unable to fully understand your concerns, But if your are taking about usage of this method, I can't find any usage of method `get_or_create_enrollment` in `post` request.